misc/dbt-materialize/dbt/include/materialize/macros/utils/generate_cluster_name.sql

```diff
@@ -13,9 +13,18 @@
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
 
-{% macro generate_cluster_name(default_cluster) -%}
+{% macro generate_deploy_cluster_name(custom_cluster_name) -%}
+    {{ return(adapter.dispatch('generate_deploy_cluster_name', 'materialize')(custom_cluster_name)) }}
+{%- endmacro %}
+
+{% macro materialize__generate_deploy_cluster_name(custom_cluster_name) -%}
+    {{ custom_cluster_name }}_dbt_deploy
+{%- endmacro %}
 
-    {%- set deploy_suffix = "_dbt_deploy" if var('deploy', False) else "" -%}
-    {{ default_cluster }}{{ deploy_suffix }}
+{% macro generate_cluster_name(custom_cluster_name) -%}
+    {{ return(adapter.dispatch('generate_cluster_name', 'materialize')(custom_cluster_name)) }}
+{%- endmacro %}
 
+{% macro materialize__generate_cluster_name(custom_cluster_name) -%}
+    {{ custom_cluster_name }}
 {%- endmacro %}
```

Comment:
As we just discussed with Nikhil, we came up with 2 suggestions here:
- Use 4 macros with the dispatch method, as how [dbt does it](https://github.com/dbt-labs/dbt-adapters/blob/a2292c8ec76e4c9f03869ad95817a2ad82dfb34b/dbt/include/global_project/macros/relations/drop.sql#L26-L34), and as currently is in this PR.
- An alternative option that we discussed is to use just two macros without using the `adapter.dispatch` method, eg:
    ```jinja
    {% macro generate_deploy_cluster_name(custom_cluster_name) -%}
        {{ custom_cluster_name }}_dbt_deploy
    {%- endmacro %}

    {% macro generate_cluster_name(custom_cluster_name) -%}
        {{ custom_cluster_name }}
    {%- endmacro %}
    ```
    
    As the macros are specific to our adapter, using just the two macros above without `adapter.dispatch`, will simplify things a bit. 

In both cases, people will be able to override this with:

```jinja
{% macro generate_cluster_name(custom_cluster_name, node) -%}

    {%- set default_cluster = target.cluster -%}
    {%- if custom_cluster_name is none -%}

        {{ default_cluster }}

    {%- else -%}

        {{target.name}}_{{ custom_cluster_name }}

    {%- endif -%}

{%- endmacro %}
```

Open to suggestions @morsapaes 🙌

Reply:
> An alternative option that we discussed is to use just two macros without using the adapter.dispatch method

This makes sense to me. We don't need to use the dispatch method, since that's only relevant for overriding core macros.

Reply:
Sounds good to me! I just updated the macros, so now we only have 2 macros left.